src/main/java/com/rtb/projectmanagementtool/user/CreateUserServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.rtb.projectmanagementtool.user;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.*;
+import com.google.appengine.api.users.*;
+import java.io.*;
+import java.util.*;
+import javax.servlet.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.*;
+
+/** Servlet that returns task data */
+@WebServlet("/create-user")
+public class CreateUserServlet extends HttpServlet {
+
+  DatastoreService datastore;
+  Cookie cookie;
+
+  public CreateUserServlet() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  // for testing
+  public CreateUserServlet(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // call auth service to generate AuthID
+    UserService userService = UserServiceFactory.getUserService();
+    String AuthID = userService.getCurrentUser().getUserId();
+
+    // initialize userMajors
+    ArrayList<String> majors = new ArrayList<>();
+
+    // create new UserData object with UserID
+    UserData newUser = new UserData("", "", 0, majors, Skills.NONE, 0);
```

Comment:
Try to move these into your controller.  The servlet (view) should only know about building the view.  The parts that interacts with the data (datastore) shouldn't be visible here.